include/triton/Dialect/TritonGPU/IR/LinearLayoutConversions.h

```diff
@@ -0,0 +1,36 @@
+// Conversions from TritonGPU layouts (e.g. BlockedEncodingAttr) to
+// LinearLayout.
+
+#include <optional>
+
+#include "triton/Tools/LinearLayout.h"
+
+namespace mlir::triton::gpu {
+
+// - BlockedEncodingAttrs have the following input dimensions.
+//
+//   "register": elements in one thread
+//   "lane": threads in a warp
+//   "warp": warps in a block/CTA
+//   "block": blocks in a cluster
+//
+// - An n-dimensional SharedEncodingAttr has the following input dimensions.
+//
+//   "dimi" for i in 0..n-1: the location in the n'th logical dimension of the
+//   input tensor.  These are not reordered according to the layout's `order`.
```

Comment:
^^ this comment needs fixing as we discussed (the input dimensions should be offset and block)